src/QsCompiler/TestTargets/Simulation/Target/QirExecution.cs

```diff
@@ -28,7 +27,42 @@ public static class JitCompilation
         private static void Main(string[] args) =>
             BuildAndRun(args[0], args[1..]);
 
-        public static void BuildAndRun(string pathToBitcode, params string[] functionNames)
+        private static unsafe bool TryParseBitcode(string path, out LLVMSharp.Interop.LLVMModuleRef outModule, out string outMessage)
+        {
+            LLVMSharp.Interop.LLVMMemoryBufferRef handle;
+            sbyte* msg;
+            if (LLVMSharp.Interop.LLVM.CreateMemoryBufferWithContentsOfFile(path.AsMarshaledString(), (LLVMSharp.Interop.LLVMOpaqueMemoryBuffer**)&handle, &msg) != 0)
+            {
+                var span = new ReadOnlySpan<byte>(msg, int.MaxValue);
+                var errTxt = span.Slice(0, span.IndexOf((byte)'\0')).AsString();
+                LLVMSharp.Interop.LLVM.DisposeMessage(msg);
+                throw new InternalCodeGeneratorException(errTxt);
+            }
+
+            fixed (LLVMSharp.Interop.LLVMModuleRef* pOutModule = &outModule)
+            {
+                sbyte* pMessage = null;
+                var result = LLVMSharp.Interop.LLVM.ParseBitcodeInContext(
```

Comment:
So you still need LLVMSharp for testing?

I think I see why -- it looks like you'd have to add a lot of functionality to the custom interop to avoid it. Just making sure that testing is the only place it's used.

Reply:
Yes, that's correct. Right now we only use LLVM JIT support in the tests, so it made sense not to include that in the built LLVM binaries. To avoid breaking the tests, I convert them to using LLVMSharp for JIT. If/when we decide to add JIT into the LLVM binaries we produce to line up with a production feature, we can switch this code to using that approach instead.